src/debug.rs

```diff
@@ -0,0 +1,274 @@
+
+        let text = format!("ID: {:?}\nLocation: x{} y{}\nPress (Primary {}, Secondary {}, Middle {})\n{}Depth: {:0.2?}\nPosition: {:0.2?}\nNormal: {:0.2?}\nInteractions: {:?}\n",
+            id,
+            position.x,
+            position.y,
+            bool_to_icon(&press.is_primary_pressed()),
+            bool_to_icon(&press.is_secondary_pressed()),
+            bool_to_icon(&press.is_middle_pressed()),
+            selection,
+            hover_data(id).map(|h| h.1.depth),
+            hover_data(id).and_then(|h| h.1.position),
+            hover_data(id).and_then(|h| h.1.normal),
+            interaction.iter()
+        );
+        use bevy_egui::egui;
+
+        let center = egui::pos2(x, y);
+
+        let dbg_painter = ctx.layer_painter(egui::LayerId::debug());
+        dbg_painter.circle(
+            center,
+            20.0,
+            egui::Color32::TRANSPARENT,
+            egui::Stroke::new(
+                3.0,
+                if press.is_any_pressed() {
+                    egui::Color32::GREEN
+                } else {
+                    egui::Color32::RED
+                },
+            ),
+        );
+
+        let new_alignment = match (left, top) {
+            (true, true) => egui::Align2::RIGHT_BOTTOM,
+            (true, false) => egui::Align2::RIGHT_TOP,
+            (false, true) => egui::Align2::LEFT_BOTTOM,
+            (false, false) => egui::Align2::LEFT_TOP,
+        };
+
+        dbg_painter.debug_text(
+            (center.to_vec2() - new_alignment.to_sign() * egui::vec2(20.0, 20.0)).to_pos2(),
+            match (alignment.to_owned(), near_border) {
+                (Some(cached), false) => cached,
+                _ => {
+                    *alignment = Some(new_alignment);
+                    new_alignment
+                }
+            },
+            egui::Color32::WHITE,
+            text,
+        );
+    }
+}
+
+#[cfg(feature = "backend_egui")]
+enum InteractionDebug {
+    Name(Name, Entity),
+    Entity(Entity),
+}
+
+#[cfg(feature = "backend_egui")]
+impl std::fmt::Debug for InteractionDebug {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        match self {
+            Self::Name(name, entity) => write!(f, "{} ({:?})", name.as_str(), entity),
+            Self::Entity(entity) => write!(f, "{entity:?}"),
+        }
+    }
+}
+
+/// Draw text on each cursor with debug info
+pub fn debug_draw(
+    mut commands: Commands,
+    asset_server: Res<AssetServer>,
+    hover_map: Res<HoverMap>,
+    pointers: Query<(
+        Entity,
+        &pointer::PointerId,
+        &pointer::PointerLocation,
+        &pointer::PointerPress,
+        &events::PointerInteraction,
+    )>,
+    #[cfg(feature = "selection")] selection: Query<Option<&selection::PointerMultiselect>>,
+    primary_window: Query<&Window, With<PrimaryWindow>>,
+) {
+    let font = asset_server.load("fonts/FiraMono-Medium.ttf");
```

Comment:
This could be configurable by user ? (I'm fine with leaving it as default)

If every plugin hardcode their debug font it could lead to a lot of arbitrary debug fonts just sitting there for debug compatibility, not very practical.

Reply:
I have no idea if there is a good solution to this, I've never seen one anyway. Suggestions welcome.

Reply:
Managed to get this working, though it's a bit cursed. https://discord.com/channels/691052431525675048/691052431974465548/1096216676460146758